sycl/include/CL/sycl/types.hpp

```diff
@@ -2399,6 +2399,12 @@ struct is_device_copyable<
     T, std::enable_if_t<std::is_trivially_copyable<T>::value>>
     : std::true_type {};
 
+// Specializations of device copyable should propagate onto constants.
+template <typename T>
+struct is_device_copyable<
+    const T, std::enable_if_t<!std::is_trivially_copyable<const T>::value>>
```

Comment:
Could you please add a comment explaining why this `enable_if` is needed?

Reply:
This has been removed.